drivers/shared/src/vic_time.c

```diff
@@ -0,0 +1,794 @@
+    }
+    else if (calendar == CALENDAR_ALL_LEAP || calendar == CALENDAR_366_DAY) {
+        _DMYAllLeap(jd, date);
+    }
+    else if (calendar == CALENDAR_360_DAY) {
+        _DMY360Day(jd, date);
+    }
+    else {
+        log_err("Unknown Calendar Flag: %hu", calendar);
+    }
+
+    return;
+}
+
+/******************************************************************************
+ * @brief   Make array of last day of months
+ * @return  Return pointer to last day of month array
+ *****************************************************************************/
+void
+make_lastday(unsigned short calendar,
+             unsigned short year,
+             unsigned short lastday[])
+{
+    size_t         i;
+    unsigned short temp[MONTHS_PER_YEAR] = {
+        31, 28, 31, 30, 31, 30,
+        31, 31, 30, 31, 30, 31
+    };
+
+    for (i = 0; i < MONTHS_PER_YEAR; i++) {
+        if (calendar == CALENDAR_360_DAY) {
+            lastday[i] = 30;
+        }
+        else {
+            lastday[i] = temp[i];
+        }
+    }
+    // leap year?
+    if (calendar == CALENDAR_JULIAN ||
+        calendar == CALENDAR_STANDARD ||
+        calendar == CALENDAR_GREGORIAN ||
+        calendar == CALENDAR_PROLEPTIC_GREGORIAN) {
+        if (_leap_year(year, calendar)) {
+            lastday[1] = 29;
+        }
+    }
+    else if (calendar == CALENDAR_366_DAY ||
+             calendar == CALENDAR_ALL_LEAP) {
+        lastday[1] = 29;
+    }
+    return;
+}
+
+/******************************************************************************
+ * @brief   Make array of last day of months
+ * @return  Return pointer to last day of month array
+ *****************************************************************************/
+bool
+_leap_year(unsigned short year,
+           unsigned short calendar)
+{
+    bool leap = false;
+
+    if ((calendar == CALENDAR_JULIAN ||
+         calendar == CALENDAR_STANDARD ||
+         calendar == CALENDAR_GREGORIAN ||
+         calendar == CALENDAR_PROLEPTIC_GREGORIAN) &&
+        (year % 4 == 0)) {
+        leap = true;
+        if ((calendar == CALENDAR_PROLEPTIC_GREGORIAN) &&
+            (year % 100 == 0) && (year % 400 != 0)) {
+            leap = false;
+        }
+        else if ((calendar == CALENDAR_STANDARD ||
+                  calendar == CALENDAR_GREGORIAN) &&
+                 (year % 100 == 0) && (year % 400 != 0) && (year < 1583)) {
+            leap = false;
+        }
+    }
+    return leap;
+}
+
+/******************************************************************************
+ * @brief   Initialize global time origin
+ *****************************************************************************/
+void
+initialize_time()
+{
+    extern global_param_struct global_param;
+
+    dmy_struct                 dmy;
+
+    // Origin is 0001-01-01
+    dmy.year = 1;
+    dmy.month = 1;
+    dmy.day = 1;
+    dmy.day_in_year = 1;
+    dmy.dayseconds = 0;
+
+    // Set origin using by using date2num with numeric origin of 0.
```

Comment:
Can you still fix the comment in line 714? As is, the line is neither english not understandable.